src/tools/debugger/nodes/BreakpointNode.java

```diff
@@ -12,16 +13,16 @@
  */
 public abstract class BreakpointNode extends AbstractBreakpointNode {
 
-  protected final BreakpointEnabling<?> breakpoint;
+  protected final BreakpointEnabling breakpoint;
 
-  protected BreakpointNode(final BreakpointEnabling<?> breakpoint) {
+  protected BreakpointNode(final BreakpointEnabling breakpoint) {
     this.breakpoint = breakpoint;
   }
 
   @Specialization(assumptions = "bpUnchanged", guards = "breakpoint.isDisabled()")
   public final boolean breakpointDisabled(
       @Cached("breakpoint.getAssumption()") final Assumption bpUnchanged) {
-    return false;
+    return ActivityThread.isSteppingType(breakpoint.getSteppingType());
```

Comment:
then the disabled state of the breakpoint will depend on the stepping. I think is a good approach. It fits better in this location than what I implemented, I think.